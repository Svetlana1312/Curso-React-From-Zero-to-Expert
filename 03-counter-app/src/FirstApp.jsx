import PropTypes from 'prop-types';


 const getMessage = () =>
 'Este es un mensaje en pantalla con anonima fuera del componente';

export const FirstApp = ({ title, subtitle, name}) => {
    //const newMessage = {
  //      message: 'Hola mundo',
   //     title: 'Svetlana'
   //  };
  // if (!title) 
   //     throw new Error('Title no esta establecido'); 
     
  return (
   <>
   <h1>{ title }</h1>
   <h2>{ subtitle }</h2>
   <p>{ name } </p>
   
   </> 
  )
}
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
   // title: 'Titulo por defecto',
    subtitle: 'Subtitulo por defecto',
    name: 'Nombre por defecto'
}
