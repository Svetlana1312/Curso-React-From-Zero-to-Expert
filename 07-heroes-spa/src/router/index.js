export * from './PrivateRouter';
export * from './AppRouter'