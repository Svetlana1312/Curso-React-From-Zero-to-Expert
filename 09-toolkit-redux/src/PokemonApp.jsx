import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon";


export const PokemonApp = () => {

    const { isLoading, pokemons, page } = useSelector((state) => state.pokemons);
   
    const dispatch = useDispatch();
    useEffect(() => {
      dispatch (getPokemons(page));
    
      
    }, []);
    if( isLoading )
       return (<h1>Loading ....</h1> );
    
    return (
        <> 
                <h1>Pokemon App</h1>
                <hr/>
                <ul>
                    {
                        pokemons.map( pokemon => 
                            <li key={ pokemon.name }>{ pokemon.name }</li>
                    )}
                    
                </ul>
                <button onClick={ () => dispatch(getPokemons( page )) }>
                    Next
                </button>
        </>)
    
  
  
}
