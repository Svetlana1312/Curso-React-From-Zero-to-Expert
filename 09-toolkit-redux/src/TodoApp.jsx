
import { useState } from "react";
import { useGetTodoQuery, useGetTodosQuery } from "./store/api"


export const TodoApp = () => {
    // const {data: todos = [], isLoading } = useGetTodosQuery();
    const [todoId, setTodoId] = useState(1);
    const {data: todo = {}, isLoading } = useGetTodoQuery(todoId);
    const onPrevClick = ()=>{
        if(todoId === 1)
            return;
       setTodoId(todoId - 1);
    }
    const onNextClick = ()=>{
         setTodoId(todoId + 1);
    }
  return (
    <>
    <h1>Todos - RTK Query</h1>
    <hr />
    <h4>Loading ....: {isLoading ? 'True' : 'False' }</h4>
    <pre>
       { JSON.stringify(todo)
       }
    </pre>
    {/* <ul>
        {todos.map((todo)=><li key={todo.id}>
    <strong> {todo.completed?'DONE: ':'Pending: '}</strong>{todo.title}
        </li>)}
    </ul> */}

    <button onClick={ onPrevClick }>Previous</button>
    <button onClick={ onNextClick }>Next</button>
    </>
  )
}
