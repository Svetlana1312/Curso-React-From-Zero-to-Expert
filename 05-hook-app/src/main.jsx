import React from 'react'
import ReactDOM from 'react-dom/client'
//import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook'
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks'
// import { Padre } from './07-tarea-memo/Padre'
import { TodoApp } from './08-useReducer/TodoApp'
import { MainApp } from './09-useContext/MainApp'
//import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
//import { SimpleForm } from './02-useEffect/SimpleForm'
//import { CounterApp } from './01-useState/CounterApp'
//import { HooksApp } from './HooksApp'
import './index.css'
//import './08-useReducer/intro-javascript'
import { BrowserRouter } from "react-router-dom";


ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
      <React.StrictMode>
        <MainApp />
   </React.StrictMode>
  
  </BrowserRouter>
 
)
