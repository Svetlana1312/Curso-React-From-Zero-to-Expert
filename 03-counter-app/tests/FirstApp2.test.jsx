import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => { 
  const title = 'Hola, soy Svetla';
    test('Debe coincidir con el snapshot', () => { 
       const { container } = render(<FirstApp title={ title } />);

       expect( container ).toMatchSnapshot();
     });

     test('Debe mostrar el titulo "Hola, soy Svetla"', () => { 
      
      render(<FirstApp title={ title }/>);

      //screen.debug();

      expect(screen.getByText(title)).toBeTruthy();
      });
    test('Debe mostrar el titulo en un h1', () => { 

      render(<FirstApp title={ title }/>);
      expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
     })
 });