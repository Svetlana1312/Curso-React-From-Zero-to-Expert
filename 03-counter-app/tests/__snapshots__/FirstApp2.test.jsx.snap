// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Svetla
  </h1>
  <h2>
    Subtitulo por defecto
  </h2>
  <p>
    Nombre por defecto
     
  </p>
</div>
`;
